core/src/main/scala/org/apache/spark/h2o/H2OContext.scala

```diff
@@ -255,7 +265,142 @@ object H2OContext extends Logging{
     * @return H2O Context
     */
   def getOrCreate(sc: SparkContext): H2OContext = {
-    getOrCreate(sc, new H2OConf(sc))
+    getOrCreate(sc, new H2OConf(sc))(SQLContext.getOrCreate(sc))
+  }
+
+  val defaultFieldNames = (i: Int) => "f" + i
```

Comment:
No no, keep h2o context here please. But `perTypeDataPartition` and related functionalities should be in its own class in `converters` package, not in H2OContext